### PR TITLE
Register .orarep file type and openra:// URI scheme on Windows

### LINKED ISSUE #7058: Command-line arguments and some sort of "URI scheme" ?
It's an Idea/Feature.
The short idea is to be able to join an OpenRA server from anywhere outside OpenRA, like adding a Join button in front of every game listed on http://www.openra.net/games/.
The long idea is to be able to create links like _openra://123.123.123.123:1234_ (I don't know if it really has a connection with http://en.wikipedia.org/wiki/URI_scheme, I never investigated it) which would be interpreted by the operating systems into something like _openra.exe -connect 123.123.123.123:1234_.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/6684006-command-line-arguments-and-some-sort-of-uri-scheme?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by penev92**
Closes #7058

### CODE REVIEW

**Comment by Mailaender** [comment]
Depends on https://github.com/OpenRA/OpenRA/pull/7934 which adds `Launch.URI`.

File: packaging/windows/OpenRA.nsi [hunk 1]
```diff
@@ -53,7 +53,21 @@ Var StartMenuFolder
...
+	; Replay file association
+	WriteRegStr HKLM "Software\Classes\.orarep" "" "OpenRA_replay"
+	WriteRegStr HKLM "Software\Classes\OpenRA_replay\DefaultIcon" "" "$INSTDIR\OpenRA.ico,0"
+	WriteRegStr HKLM "Software\Classes\OpenRA_replay\Shell\Open\Command" "" "$INSTDIR\OpenRA.exe Launch.Replay=%1"
+	
+	; OpenRA URL Scheme
+	WriteRegStr HKLM "Software\Classes\openra" "" "URL:OpenRA scheme"
+	WriteRegStr HKLM "Software\Classes\openra" "URL Protocol" ""
+	WriteRegStr HKLM "Software\Classes\openra\DefaultIcon" "" "$INSTDIR\OpenRA.ico,0"
+	WriteRegStr HKLM "Software\Classes\openra\Shell\Open\Command" "" "$INSTDIR\OpenRA.exe Launch.URI=%1"
```
Comment (obrakmann) [inline packaging/windows/OpenRA.nsi]: Do you have to add these in the `Clean` section, too, so that the uninstaller will properly remove them?

Comment (penev92) [inline packaging/windows/OpenRA.nsi]: Well maybe I should.

**Comment by penev92** [comment]
Added registry cleanup on uninstall.

**Comment by obrakmann** [comment]
From my understanding of NSIS, this looks correct. :+0.5:, since I cannot test it.

Needs testing from a Windows user. Penev, maybe you could upload the installer you created to help  reviewers.

**Comment by RoosterDragon** [comment]
If I install & uninstall it, it leaves a bunch of dead shortcuts in my start menu.

File: OpenRA.GameMonitor/GameMonitor.cs [hunk 2]
```diff
@@ -20,12 +20,15 @@ namespace OpenRA
 {
 	class GameMonitor
 	{
-		static string processName = "OpenRA.Game.exe";
 		static Process gameProcess;
 
 		[STAThread]
 		static void Main(string[] args)
 		{
+			var executableDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().GetName().CodeBase);
```
Comment (RoosterDragon) [inline OpenRA.GameMonitor/GameMonitor.cs]: ```Assembly.GetExecutingAssembly().Location``` might let you skip having to remove the ```file``` protocol below.

Comment (penev92) [inline OpenRA.GameMonitor/GameMonitor.cs]: Thanks. Changed.

**Comment by RoosterDragon** [comment]
This cleans up the registry keys fine (the ones it is targeting for removal, anyway), and double clicking on a replay to play it works too.

**Comment by penev92** [comment]
I'll take this as a :+0.75:

File: packaging/windows/OpenRA.nsi [hunk 3]
```diff
@@ -214,7 +228,12 @@ Function ${UN}Clean
 	Delete $INSTDIR\freetype6.dll
 	Delete $INSTDIR\SDL2-CS.dll
 	RMDir /r $INSTDIR\Support
+	
 	DeleteRegKey HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\OpenRA"
+	DeleteRegKey HKLM "Software\Classes\.orarep"
+	DeleteRegKey HKLM "Software\Classes\.OpenRA_replay"
```
Comment (Mailaender) [inline packaging/windows/OpenRA.nsi]: There is no leading dot in the install section.

Comment (penev92) [inline packaging/windows/OpenRA.nsi]: Nice catch!

**Comment by penev92** [comment]
Fixed. Should now delete all registry.

**Comment by Mailaender** [comment]
Thanks! This completes our shell integration nicely.

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/d49882200e5762d26480fc40cd7bcae72d165f48...b8f4a2c4ad688db8b5646c97b26ec819e27d78d8)

**Comment by x-a-n-a-x** [comment]
Clicking on the link (http://www.openra.net/games/) works fine, it launches OpenRA ; but here is what I see:
appassociationlaunch

The logo and application name (OpenRA) are missing.

**Pull request merged**